packages/beacon-node/src/network/reqresp/beaconBlocksMaybeBlobsByRange.ts

```diff
@@ -105,7 +105,8 @@ export async function beaconBlocksMaybeBlobsByRange(
 
     const dataColumnRequest = {...request, columns};
     const [allBlocks, allDataColumnSidecars] = await Promise.all([
-      partialDownload
+      // TODO-das: investigate why partialDownload blocks is empty here
```

Comment:
this is because when we pick a peer for a batch we don't check if target slot of peer is higher than batch's start slot
https://github.com/ChainSafe/lodestar/blob/27365a35fd8bf28d37c8e9e8393c0b690f052915/packages/beacon-node/src/sync/range/utils/peerBalancer.ts#L64

we should also make sure the peer has columns we need